conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java

```diff
@@ -83,6 +83,13 @@ public static <T> void addAll(Collection<T> addTo, Iterable<? extends T> element
 
     @SuppressWarnings("unchecked")
     public static <T> void addAllAndCheckNonNull(Collection<T> addTo, Iterable<? extends T> elementsToAdd) {
+        if (elementsToAdd instanceof ConjureIntegerList
```

Comment:
It feels a bit weird to have this method be named `addAllAndCheckNonNull` and have it _not_ check that elements are non-null. It'd be nice if we can change the callers of `addAllAndCheckNonNull` to use `addAll` if we're using primitive collections.